packages/config-array/src/config-array.js

```diff
@@ -244,6 +244,80 @@ function doMatch(filepath, pattern, options = {}) {
 	return matcher.match(filepath);
 }
 
+/**
+ * Normalizes a pattern by removing the leading "./" if present.
+ * @param {string} pattern The pattern to normalize.
+ * @returns {string} The normalized pattern.
+ */
+function normalizePattern(pattern) {
+	if (isString(pattern)) {
+		if (pattern.startsWith("./")) {
+			return pattern.slice(2);
+		}
+
+		if (pattern.startsWith("!./")) {
+			return `!${pattern.slice(3)}`;
+		}
+	}
+
+	return pattern;
+}
+
+/**
+ * Checks if a given pattern requires normalization.
+ * @param {any} pattern The pattern to check.
+ * @returns {boolean} True if the pattern needs normalization, false otherwise.
+ *
+ */
+function needsPatternNormalization(pattern) {
+	return (
+		isString(pattern) &&
+		(pattern.startsWith("./") || pattern.startsWith("!./"))
+	);
+}
+
+/**
+ * Normalizes `files` and `ignores` patterns in a config by removing "./" prefixes.
+ * @param {Object} config The config object to normalize patterns in.
+ */
+function normalizeConfigPatterns(config) {
```

Comment:
```suggestion
function normalizeConfigPatterns(config) {
	if (!config) {
		return config;
	}
```

I've just tried this version of config-array with eslint/eslint, and several tests that check how nullish config elements are reported are failing with errors like `Cannot read properties of undefined (reading 'files')`. I think we can address this by adding an early exit for that case.

Reply:
Can you also add a test with `null` and `undefined` configs in this PR?